models/oneinch/fantom/oneinch_fantom_calls_transfers.sql

```diff
@@ -0,0 +1,126 @@
+{{ config( 
+    alias = alias('fantom_calls_transfers'),
+    tags = ['dunesql'],
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['tx_hash', 'start', 'transfer_trace_address']
+    )
+}}
+
+{% set project_start_date = "timestamp '2020-11-13'" %} 
+{% set lookback_days = -2 %}
+{% set transfer_selector = '0xa9059cbb' %}
+{% set transfer_from_selector = '0x23b872dd' %}
+{% set selector = 'substr(input, 1, 4)' %}
+
+with
+
+calls as (
+    select 
+        tx_hash
+        , gr_traces.start
+        , gr_traces.caller
+        , transactions.tx_from
+        , transactions.tx_success
+        , gr_traces.call_success
+        , gr_traces.call_selector
+    from (
+        select 
+            "from" as tx_from
+            , "hash" as tx_hash
+            , success as tx_success
+            , block_time from {{ source('fantom', 'transactions') }}
+        where
+        {% if is_incremental() %}
+            block_time >= date_add('day', {{lookback_days}}, now())
+        {% else %}
+            block_time >= {{ project_start_date }}
+        {% endif %}
+    ) as transactions 
+    join (
+        select 
+            tx_hash
+            , min(trace_address) as start
+            , min_by("from", trace_address) as caller
+            , min_by(success, trace_address) as call_success
+            , min_by({{selector}}, trace_address) as call_selector
+        from {{ source('fantom', 'traces') }}
+        where 
+            {% if is_incremental() %}
+                block_time >= date_add('day', {{lookback_days}}, now())
+            {% else %}
+                block_time >= {{ project_start_date }}
+            {% endif %}
+            and "to" in (
```

Comment:
you may get better performance switching this to an inner join instead, just a thought

Reply:
As far as I know from the experience with other columnar databases, inner-join is more effective than semi-join in cases when there are lot of rows in the second table. In this particular case there are just 30 addresses, so it shouldn't make a difference

btw tested both cases, ran at the same time

large cluster
<img width="1400" alt="image" src="https://github.com/duneanalytics/spellbook/assets/40689054/2ac3718c-ea31-4310-9ab5-19b6f99a2767">

medium cluster
<img width="1412" alt="image" src="https://github.com/duneanalytics/spellbook/assets/40689054/e12b3081-72a9-4ca5-9b39-6d8965fd592e">

I think it won't make a difference, wdyt?

Reply:
thanks for the evidence -- i'm good with as-is